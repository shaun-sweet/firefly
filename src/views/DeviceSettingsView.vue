<template>
  <!-- if you want automatic padding use "layout-padding" class -->
  <div class="layout-padding">
    <q-modal ref="addNodeMenu">
      <div class="layout-padding">
        <h4>
          Add Node
        </h4>
        <q-btn color="primary" @click="$refs.addNodeMenu.close()">Close</q-btn>
      </div>
    </q-modal>
    <q-modal ref="addPresenceDeviceMenu">
      <div class="layout-padding">
        <h4>Add Presence Device</h4>
        <q-btn color="primary" @click="$refs.addPresenceDeviceMenu.close()">Close</q-btn>
      </div>
    </q-modal>
    <div class="button-group">
      <q-btn color="primary" @click.prevent="handleAddZwave" big icon="ion-plus">Add Zwave Node</q-btn><br />
      <q-btn color="faded" @click.prevent="handleRemoveZwave" big icon="ion-android-cancel">Remove Zwave Node</q-btn><br />
      <q-btn color="negative" @click.prevent="handleCancelZwave" big icon="ion-toggle-filled">Cancel Zwave Command</q-btn><br />
      <q-btn color="amber" @click.prevent="handleRepairZwave" big icon="ion-toggle-filled">Repair Zwave Command</q-btn>
    </div>

  </div>
</template>

<script>
import * as firebase from 'src/firebase'

export default {
  data () {
    return {

    }
  },
  methods: {
    handleAddZwave () {
      const state = this.$store.state
      const payload = {
        homeId: state.selectedHome,
        command: 'add_node'
      }
      firebase.zWaveCommand(payload)
    },

    handleRemoveZwave () {
      const state = this.$store.state
      const payload = {
        homeId: state.selectedHome,
        command: 'remove_node'
      }
      firebase.zWaveCommand(payload)
    },

    handleCancelZwave () {
      const state = this.$store.state
      const payload = {
        homeId: state.selectedHome,
        command: 'cancel'
      }
      firebase.zWaveCommand(payload)
    },
    handleRepairZwave () {

    }
  }
}
</script>

<style lang="stylus">
.button-group button
  margin-bottom 1em

</style>
