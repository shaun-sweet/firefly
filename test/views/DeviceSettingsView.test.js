import Vue from 'vue';
import { shallow } from 'vue-test-utils';
import DeviceSettingsView from 'src/views/DeviceSettingsView.vue';

describe('DeviceSettingsView', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(DeviceSettingsView);
  })
  it('should render the view correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it('should call handleAddZwave when Add Zwave clicked', () => {
    const spy = jest.fn();
    wrapper.vm.handleAddZwave = spy;
    wrapper.update()
    const addButton = wrapper.findAll('q-btn').at(2);
    addButton.trigger('click');
    expect(spy).toHaveBeenCalled();
  });
  it('should call handleRemoveZwave when Add Zwave clicked', () => {
    const spy = jest.fn();
    wrapper.vm.handleRemoveZwave = spy;
    wrapper.update()
    const removeButton = wrapper.findAll('q-btn').at(3);
    removeButton.trigger('click');
    expect(spy).toHaveBeenCalled();
  });
  it('should call handleCancelZwave when Add Zwave clicked', () => {
    const spy = jest.fn();
    wrapper.vm.handleCancelZwave = spy;
    wrapper.update()
    const cancelButton = wrapper.findAll('q-btn').at(4);
    cancelButton.trigger('click');
    expect(spy).toHaveBeenCalled();
  });
  it('should call handleRepairZwave when Add Zwave clicked', () => {
    const spy = jest.fn();
    wrapper.vm.handleRepairZwave = spy;
    wrapper.update()
    const repairButton = wrapper.findAll('q-btn').at(5);
    repairButton.trigger('click');
    expect(spy).toHaveBeenCalled();
  });

})
