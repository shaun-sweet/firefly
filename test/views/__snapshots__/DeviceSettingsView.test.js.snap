// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceSettingsView should render the view correctly 1`] = `
<div
  class="layout-padding"
>
  <q-modal>
    <div
      class="layout-padding"
    >
      <h4>
        
        Add Node
      
      </h4>
       
      <q-btn
        color="primary"
      >
        Close
      </q-btn>
    </div>
  </q-modal>
   
  <q-modal>
    <div
      class="layout-padding"
    >
      <h4>
        Add Presence Device
      </h4>
       
      <q-btn
        color="primary"
      >
        Close
      </q-btn>
    </div>
  </q-modal>
   
  <div
    class="button-group"
  >
    <q-btn
      big=""
      color="primary"
      icon="ion-plus"
    >
      Add Zwave Node
    </q-btn>
    <br />
     
    <q-btn
      big=""
      color="faded"
      icon="ion-android-cancel"
    >
      Remove Zwave Node
    </q-btn>
    <br />
     
    <q-btn
      big=""
      color="negative"
      icon="ion-toggle-filled"
    >
      Cancel Zwave Command
    </q-btn>
    <br />
     
    <q-btn
      big=""
      color="amber"
      icon="ion-toggle-filled"
    >
      Repair Zwave Command
    </q-btn>
  </div>
</div>
`;
